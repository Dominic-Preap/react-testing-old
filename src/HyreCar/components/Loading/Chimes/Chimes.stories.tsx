import * as React from 'react';

import { storiesOf } from '@storybook/react';

import { Chimes } from './Chimes';

storiesOf('HyreCar|Components/Loading', module).add('Chimes', () => <Chimes />);
