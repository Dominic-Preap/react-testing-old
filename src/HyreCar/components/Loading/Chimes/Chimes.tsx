import * as React from 'react';

import Typography from '@material-ui/core/Typography';
import styled, { keyframes } from 'styled-components';

interface SpanProps {
  value: number;
}

const rotate = keyframes`
  0% {
    transform: rotateZ(-20deg);
  }
  50% {
    transform: rotateZ(20deg) scaleY(1.2);
    background: rgb(163, 231, 252);
  }
  100% {
    transform: rotateZ(-20deg);
  }
`;

const LoadingContainer = styled.div`
  margin: 100px auto;
  text-align: center;
`;
const LoadingText = styled.div`
  margin-top: 40px;
`;

const LoadingWrapper = styled.div`
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  margin: 0 auto;
  position: relative;
  width: 234px;
`;

const BaseSpan = styled.span`
  background-clip: padding-box;
  background: rgb(0, 157, 236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  transform-origin: 50% 0%;
  width: 5.5px;
`;

// https://github.com/styled-components/styled-components/issues/630#issuecomment-462907177
const Span = styled(BaseSpan)<SpanProps>`
  animation: ${rotate} 1.45s ${props => `${0.07 + 0.08 * +props.value}s`} infinite cubic-bezier(0.005, 0.56, 0.58, 1.59);
`;

// TODO:
export const Chimes = () => {
  return (
    <LoadingContainer>
      <LoadingWrapper>
        {[...Array(10).keys()].map(i => (
          <Span value={i} key={i} />
        ))}
      </LoadingWrapper>
      <LoadingText>
        <Typography variant="h5" gutterBottom={true}>
          ...Loading Results
        </Typography>
        <Typography variant="body1" gutterBottom={true}>
          Please do not refresh the page
        </Typography>
      </LoadingText>
    </LoadingContainer>
  );
};
