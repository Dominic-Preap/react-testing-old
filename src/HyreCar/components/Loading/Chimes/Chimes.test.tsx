import * as React from 'react';
import { create } from 'react-test-renderer';

import { Chimes } from './Chimes';

describe('Chimes component', () => {
  it('should render properly', () => {
    expect(create(<Chimes />).toJSON()).toMatchSnapshot();
  });
});
