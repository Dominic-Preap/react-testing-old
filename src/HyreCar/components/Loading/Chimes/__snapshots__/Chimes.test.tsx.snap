// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chimes component should render properly 1`] = `
.c0 {
  margin: 100px auto;
  text-align: center;
}

.c12 {
  margin-top: 40px;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  position: relative;
  width: 234px;
}

.c2 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.07s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.07s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c3 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.15000000000000002s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.15000000000000002s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c4 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.23s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.23s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c5 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.31s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.31s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c6 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.39s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.39s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c7 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.47000000000000003s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.47000000000000003s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c8 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.55s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.55s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c9 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.6300000000000001s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.6300000000000001s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c10 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.71s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.71s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

.c11 {
  background-clip: padding-box;
  background: rgb(0,157,236);
  border-radius: 1063px;
  display: inline-block;
  height: 64px;
  margin: 0 5.5px;
  -webkit-transform-origin: 50% 0%;
  -ms-transform-origin: 50% 0%;
  transform-origin: 50% 0%;
  width: 5.5px;
  -webkit-animation: ehCYUl 1.45s 0.79s infinite cubic-bezier(0.005,0.56,0.58,1.59);
  animation: ehCYUl 1.45s 0.79s infinite cubic-bezier(0.005,0.56,0.58,1.59);
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
      value={0}
    />
    <span
      className="c3"
      value={1}
    />
    <span
      className="c4"
      value={2}
    />
    <span
      className="c5"
      value={3}
    />
    <span
      className="c6"
      value={4}
    />
    <span
      className="c7"
      value={5}
    />
    <span
      className="c8"
      value={6}
    />
    <span
      className="c9"
      value={7}
    />
    <span
      className="c10"
      value={8}
    />
    <span
      className="c11"
      value={9}
    />
  </div>
  <div
    className="c12"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      ...Loading Results
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    >
      Please do not refresh the page
    </p>
  </div>
</div>
`;
